mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -71,6 +73,20 @@ data class MPSNode(val node: SNode) : IDefaultNodeAdapter {
             return node.children.map { MPSNode(it) }
         }
 
+    override fun replaceNode(concept: ConceptReference): INode {
+        val id = node.nodeId
+        val model = checkNotNull(node.model) { "Node is not part of a model" }
+        val newNode = model.createNode(SConceptAdapterById(SConceptId.deserialize(concept.uid), ""), id)
+        node.properties.forEach { newNode.setProperty(it, node.getProperty(it)) }
+        node.references.forEach { newNode.setReference(it.link, it.targetNodeReference) }
+        node.children.forEach { child -> child.containmentLink?.let { newNode.addChild(it, child) } }
```

Comment:
I'm not completely sure on what a `containmentLink` is. But right now, this would only add the child to the new node if that link is not null. Is that the correct logic?

Reply:
That should be fine. `containmentLink` is only null iff the node has no parent, which shouldn't happen here.
I could change it to checkNotNull() to provide an exception instead.